public_html/admin/controller/responses/setting/setting_quick_form.php

```diff
@@ -240,40 +241,30 @@ public function quick_start_save_next()
                 $this->loadLanguage('setting/setting');
                 $this->loadLanguage('common/header');
                 $this->loadLanguage('common/quick_start');
+                $post = $this->request->post;
+                $section = $group;
 
                 if ($group == 'appearance') {
                     $section = $this->request->get['tmpl_id'] == 'default'
-                        ? 'appearance'
-                        : $this->request->get['tmpl_id'];
-                    if (has_value($this->request->post['config_logo'])) {
-                        $this->request->post['config_logo'] = html_entity_decode(
-                            $this->request->post['config_logo'],
-                            ENT_COMPAT,
-                            'UTF-8'
-                        );
-                    } elseif (!$this->request->post['config_logo']
-                        && isset($this->request->post['config_logo_resource_id'])
-                    ) {
-                        //we save resource ID vs resource path
-                        $this->request->post['config_logo'] = $this->request->post['config_logo_resource_id'];
+                                ? 'appearance'
+                                : $this->request->get['tmpl_id'];
+
+                    foreach (['config_logo', 'config_mail_logo', 'config_icon'] as $n) {
+                        $newName = in_array($n, ['config_logo', 'config_mail_logo']) ? $n.'_'.$languageId : $n;
+                        if (isset($post[$n.'_resource_id'])) {
+                            //we save resource ID vs resource path
+                            $post[$newName] = $post[$n.'_resource_id'];
```

Comment:
1. We should not change POST. We talked about it. Even if we have it in the old code, we should go away from this. 
2. I do not think global post will be changed as this is a local copy.

Reply:
1. post is not changing here. see line 244. $post variable is not a reference. it's a separate var!
2. see above!